WorkspaceServer.Tests/Kernel/ObservableExtensions.cs

```diff
@@ -12,43 +12,62 @@ namespace WorkspaceServer.Tests.Kernel
 {
     public static class ObservableExtensions
     {
-        public static IObservable<JObject> TakeUntilCommandFailed(this IObservable<JObject> source)
+
+        public static IObservable<JObject> TakeUntilCommandFailed(this IObservable<string> source, TimeSpan? timeout = null)
         {
-            var termination = new Subject<Unit>();
-            return source.TakeUntil(termination)
-                .Do(e =>
-                {
-                    if (e["eventType"].Value<string>() == nameof(CommandFailed))
-                    {
-                        termination.OnNext(Unit.Default);
-                    }
-                });
+            return source
+                .Where(s => !string.IsNullOrWhiteSpace(s))
+                .Select(JObject.Parse)
+                .TakeUntilCommandFailed(timeout);
         }
 
-        public static IObservable<JObject> TakeUntilCommandHandled(this IObservable<JObject> source)
+        public static IObservable<JObject> TakeUntilCommandFailed(this IObservable<JObject> source, TimeSpan? timeout = null)
         {
-            var termination = new Subject<Unit>();
-            return source.TakeUntil(termination)
-                .Do(e =>
-                {
-                    if (e["eventType"].Value<string>() == nameof(CommandHandled))
-                    {
-                        termination.OnNext(Unit.Default);
-                    }
-                });
+            return source.TakeUntilEvent<CommandFailed>(timeout);
+        }
+
+        public static IObservable<JObject> TakeUntilCommandHandled(this IObservable<string> source, TimeSpan? timeout = null)
+        {
+            return source
+                .Where(s => !string.IsNullOrWhiteSpace(s))
+                .Select(JObject.Parse)
+                .TakeUntilCommandHandled(timeout);
+        }
+
+        public static IObservable<JObject> TakeUntilEvent<T>(this IObservable<string> source, TimeSpan? timeout = null) where T : IKernelEvent
+        {
+            return source
+                .Where(s => !string.IsNullOrWhiteSpace(s))
+                .Select(JObject.Parse)
+                .TakeUntilEvent<T>(timeout);
         }
 
-        public static IObservable<JObject> TakeUntilCommandParseFailure(this IObservable<JObject> source)
+        public static IObservable<JObject> TakeUntilEvent<T>(this IObservable<JObject> source, TimeSpan? timeout = null) where T : IKernelEvent
         {
             var termination = new Subject<Unit>();
-            return source.TakeUntil(termination)
+            var stream =  source.TakeUntil(termination)
                 .Do(e =>
                 {
-                    if (e["eventType"].Value<string>() == nameof(CommandParseFailure))
+                    if (e["eventType"].Value<string>() == typeof(T).Name)
                     {
                         termination.OnNext(Unit.Default);
                     }
                 });
+            if (timeout != null)
```

Comment:
Since these are test helpers, it's simpler to not have to specify a timeout in all of the tests, and if `null` is passed, pass a default here. There's no valid use case for not having a timeout, except...

An additional nice-to-have there is to not pass the timeout if the debugger is attached.